src/ECL/ECL.hs

```diff
@@ -68,23 +104,30 @@ genericMerge f bsl@((s, b):bs) bsr@((s', b'):bs')
 
 -- | Returns the uncovered set after one binder is applied to the set of
 -- values represented by another.
-missingSingle :: (Eq lit) => Binder lit -> Binder lit -> Binders lit
-missingSingle _ (Var _) = []
-missingSingle b@(Var _) (Tagged tag bc) =
-  map (Tagged tag) $ missingSingle b bc
-missingSingle c@(Tagged tag b) (Tagged tag' b')
-  | tag == tag' = map (Tagged tag) $ missingSingle b b'
-  | otherwise = [c]
-missingSingle (Var _) (Record bs) =
-  map (Record . zip names) miss
+missingSingle :: (Eq lit) => Environment -> Binder lit -> Binder lit -> (Binders lit, Maybe Bool)
+missingSingle _ _ (Var _) = ([], pure True)
+missingSingle env (Var _) cb@(Tagged _ _) =
+  (concatMap (\cp -> fst $ missingSingle env cp cb) $ tagEnv env, pure True)
   where
-  miss = getUncovered $ missingMultiple (initialize $ length bs) binders
+  -- arity 1
+  tag :: (Eq lit) => (Name, Arity) -> Binder lit
+  tag (n, _) = Tagged n wildcard
+
+  tagEnv :: (Eq lit) => Environment -> Binders lit
```

Comment:
Use the list type here. We used `Binders` for a product of binders, and `[]` for alternatives, and this is a set of alternatives. Same above actually in the type of `missingSingle`.